solr/core/src/java/org/apache/solr/cloud/api/collections/CategoryRoutedAlias.java

```diff
@@ -117,15 +117,15 @@ public void validateRouteValue(AddUpdateCommand cmd) throws SolrException {
 
     String dataValue = String.valueOf(cmd.getSolrInputDocument().getFieldValue(getRouteField()));
     String candidateCollectionName = buildCollectionNameFromValue(dataValue);
-    List<String> colls = getCollectionList(this.parsedAliases);
+    List<String> cols = getCollectionList(this.parsedAliases);
 
-    if (colls.contains(candidateCollectionName)) {
+    if (cols.contains(candidateCollectionName)) {
       return;
     }
 
-    if (colls.stream()
+    if (cols.stream()
         .filter(x -> !x.contains(UNINITIALIZED)).count() >= Integer.valueOf(maxCardinality)) {
-      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "max cardinality can not be exceeded for a Category Routed Alias: " + maxCardinality);
```

Comment:
Actually, Bad request is better. The server is not broken, it is correctly refusing data that it is configured to refuse. I'll tweak